### DISCUSSION TITLE
Some suggestions

### DISCUSSION THREAD

**Discussion by cbelena** [jpanther/congo#88] [Ideas]
Hi! I'm also currently creating my new personal website with Congo and I have some suggestions:

#### Regarding `layouts/partials/meta/date.html`:
1. Use `i18n` function to be able to translate the word "Updated".
2. Use [`time.Format` with date/time formatting layouts](https://gohugo.io/functions/dateformat/#datetime-formatting-layouts) instead of `.Date.Format` to output localized dates. That way you can use for example `article.dateFormat = ":date_long"` in `params.toml` to output `22 de enero de 2022` if language is set to spanish. This requires to bump minimum version of Hugo to v0.87 instead of v0.86.1, but I think it's worth it.

#### About `layouts/partials/article-pagination.html`:
1. Maybe it's a cultural thing, but I find odd that in order to navigate to older posts, I have to click on the right side of the pagination, and clicking on the left side goes to a newer post. When I want to 'go back' I always think of a leftwards arrow. So, maybe the pagination links are reversed?
2. When an article is updated and `showDateUpdated` is `true`, it shows the updated date on pagination, like this:
![image](https://user-images.githubusercontent.com/97797388/150656625-622cdad3-d52e-4659-a190-f88e3a835ab3.png)
I think the updated part is not necessary here, and also shows some weird double spaces.
To fix this, I have thought about changing `layouts/partials/meta/date.html` to only output the original date and creating a new partial called `date-updated.html` which adds the 'Updated' localized string and then calls the `date.html` partial.
And this leads me to...

#### Improving `layouts/partials/article-meta.html`:
1. Instead of showing the last updated date [in parentheses](https://github.com/jpanther/congo/discussions/43#discussioncomment-1741124), treat it like another meta parameter, such as `showWordCount` or `showReadingTime`:
![image](https://user-images.githubusercontent.com/97797388/150657867-9312f8dd-9950-4732-9fb0-bfd7ddb870c2.png)
What do you think?
2. I know this is a CSS thing, but on mobile, when there is too much text for the page width, it renders like this:
![image](https://user-images.githubusercontent.com/97797388/150658181-c4742215-8e5a-4634-ab99-6120debcd95b.png)
I have fixed it like this in `custom.css`, but I don't know if this it's the best way:
    ```
    .flex.flex-row.items-center {
      flex-wrap: wrap;
    }
    ```
    Now the text wraps like this:
    ![image](https://user-images.githubusercontent.com/97797388/150658272-5b775089-4225-4192-89bd-50d5195a8812.png)
This happens in articles and article listings (as shown above).

May I submit some PR with these changes I have already implemented?

#### Other
1. I don't know if it's a bug or feature, but if an article has `externalUrl` set, you can access the page of the 'article' through the navigation links. Maybe checking `.Params.externalUrl` is needed, like in `layouts/partials/article-link.html`?
2. In the 404 page, the `<title>` is not localized: appears as "404 Page not found · `.Site.Title`" regardless of the language. I'm not exactly sure where to put the `i18n` function to properly fix it.
3. It seems that `enableEmoji = false` and [`paginate`](https://gohugo.io/getting-started/configuration/#paginate) settings in `config.toml` don't really change anything, but I think this is pretty minor as the default value for `paginate` is OK (10) and emoji are always welcome! :grinning:
4. Currently, if `darkMode` is not set to `"auto"`, `darkToggle` doesn't do anything. I think it would be great if you can force light or dark appearance with `darkMode`, and also allowing to set `darkToggle` to `"true"` to show the toggle.
5. Speaking of `darkToggle`, could you please center the vertical alignment of the icon? :pray: It's slightly down:
![image](https://user-images.githubusercontent.com/97797388/150659598-e80fc826-a6d1-478d-aeb7-64e09d57a05b.png)
6. I would [also](https://github.com/jpanther/congo/discussions/43#discussioncomment-1852180) love to see some improvements on the main menu on mobile. Some hamburger menu like @AntoineSoetewey said would be great!

Thank you @jpanther for this **great** Hugo theme!

**Comment by jpanther** [comment]
Wow @cbelena, thanks for the really detailed suggestions. There's a lot of great changes here and I largely agree with most of what you have suggested. My only issue at the moment is that I'm very close to releasing version 2 of the theme which has a lot of changes - some of which will address some of these items. I'm hoping to avoid adding feature changes to the 1.x branch until v2 is out as some of the suggested changes here will cause substantial conflicts.

That said, if you want to submit a PR that adds the items below, I'd be happy to merge that in as I can then pull it up into v2 also...

- a new `article.updated` i18n string for the meta "Updated: " date prefix
- adjusting the metadata output to remove the parenthesis on the updated string (I'm happy to fix the CSS if you need help with that bit)

To address some of your other points, I think I'll experiment with swapping the pagination direction for v2 as it's a change that can fit in with all the other adjustments that are happening without annoying anyone who is used to the current way it works. Now that I think of it I'm not even sure why I paginated them in the direction that I did because on reflection it doesn't even make sense to me!! The `time.Format` change is already in v2 along with a whole range of i18n and language improvements.

With regards to the "other" list...

1. This is a bug. Feel free to open an issue, or you can include it in your PR if you want to fix it.
2. I think this is a Hugo oddity because of the way it treats the 404 page as a node rather than a page. You should be able to localise this string by adding a file at `content/404/_index.md` and then providing `title` in the front matter.
3. I recall that `paginate` was changing the number of items per page. Are you saying adjusting this setting isn't doing anything? The `enableEmoji` flag only affects converting when using emoji codes like `:parachute:`. It won't have any affect if you insert the emoji as a unicode symbol 🪂.
4. `darkMode` works this way so you can force the appearance and not rely on JavaScript.
5. The toggle button has completely changed in v2 so this shouldn't be an issue. But if you want to add this to your PR I can add a fix for it now too.
6. This is still on the to-do list. I have a personal dislike for mobile menus but I realise a lot of people are asking for this so I'll try to add something once I get the other things out the door. My original intent for the main menu was that it would only have a couple of links but from seeing what people have been creating, there's a need to support more detailed menus (and maybe even sub menus). I need to think more about this feature before I implement a solution.

I typed this response pretty quickly but I hope I covered all your items. Feel free to clarify anything!

**Reply by cbelena** [reply]
Thank you for your quick and complete reply!

I have submitted a PR about those two topics. I was not sure if I should have included the `time.Format` in `date.html`, but let me know what do you think :grinning:

I didn't know v2 was about to be finished soon, I will wait patiently to check out what's new!

Regarding my *other* list:

1. Ok, I will try to fix it and open a new PR; if not I will open an issue.
2. That's a great idea, thanks! I have solved it with a new custom layout and tweaking the front matter: https://github.com/cbelena/cbelena.es/commit/0fb708da0c3008f7dfc76e7c9843b5262247e5b3 Now it also shows recent articles :grinning:
3. I have tested the `paginate` setting again and you are right: it works as intended, sorry. I don't know why did not last time I checked it. About `enableEmoji = false` in `config.toml`, I have tested it again and keeps showing the emojis in content (those that are in `:parachute:` format); surely I'm doing something wrong but it really doesn't bother me: I like them :smile:
4. I understand that currently it's not possible, but just for clarify, if I want to set `darkMode = "true"`, it's not possible to also allow to change manually the appearance with the `darkToggle` button? Or this can't be done without JavaScript? I've been messing around with `footer.html` without much success haha :sweat_smile:
5. Great, I'll wait to see how it works in v2 :blush:
6. I understand your position, take all the time you need!

Thank you again!

**Reply by jpanther** [reply]
I've just done a bit more reading on `enableEmoji` and I think I've been misunderstanding how this works. I _think_ what it is intended to do is when enabled it changes the `:parachute:` to emoji in `.Content` even when it's not passed to `emojify`, however the inverse is not true - ie. if you use `emojify` and `enableEmoji = false` then it doesn't stop the conversion as `emojify` is always run. So I guess this is a pointless config value now that most things are going through `emojify` anyway.

The problem with the default behaviour is that it's only run on `.Content` and not on other values like `.Title` so you end up having to pass some through `emojify` and others not so it's not really a useful thing to be able to configure. It's only really useful for people who want to write things like `:parachute:` in their content and disable the default Hugo behaviour of having that converted automatically. No one has complained about this though so I'm assuming I can continue to get away with emojifying everything for now!

The `darkMode` behaviour you're seeking is not currently possible because when you set `darkMode = true` then that will force the appearance without the toggle. I do it this way so that you can force the appearance and not have JavaScript running on your site. I think I'll just keep adding more flexibility to this over time but this is one of those classic features where you can ask 10 people how they want it to behave and you'll get 10 different answers. To put it another way, it's not a feature I don't want to include. It just hasn't been a high priority at the moment because I think the current functionality is already quite flexible. I would have thought the vast majority of people are happy just going with their OS preference anyway. How many times would people go to someone's website and look for an appearance toggle? 🤷‍♂️

**Reply by cbelena** [reply]
Well, I have learned something else too. And I fully understand your reasoning about the `darkMode` behaviour :thumbsup:

Thank you very much for your detailed responses!

**Reply by AntoineSoetewey** [reply]
Dear @jpanther,

Sorry to come back with this: do you still expect to include a hamburger menu on mobile or it is not in the pipeline anymore?

Just asking before selecting a Hugo theme (my client requires this type of menu on mobile).

Regards,
Antoine
